File: app/src/main/java/com/micklab/llmdemo/MainActivity.java
package com.micklab.llmdemo;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        TextView tv = new TextView(this);
        tv.setText("Hello from llmdemo!");
        tv.setTextSize(24);

        setContentView(tv);
    }
}

File: app/src/main/java/com/micklab/llmdemo/MainActivity.kt
package com.micklab.llmdemo

import android.os.Bundle
import android.view.View
import android.view.ViewGroup.LayoutParams.MATCH_PARENT
import android.widget.FrameLayout
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.commit

class MainActivity : FragmentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val container = FrameLayout(this).apply {
            id = View.generateViewId()
            layoutParams = FrameLayout.LayoutParams(MATCH_PARENT, MATCH_PARENT)
        }

        setContentView(container)

        if (savedInstanceState == null) {
            supportFragmentManager.commit {
                replace(container.id, VisualizerFragment())
            }
        }
    }
}
